pkg/platform/runtime/setup/setup.go

```diff
@@ -232,17 +251,23 @@ func (s *Setup) deleteOutdatedArtifacts(changeset artifact.ArtifactChangeset, st
 	return nil
 }
 
-func (s *Setup) installFromBuildResult(buildResult *model.BuildResult, _ map[artifact.ArtifactID]artifact.ArtifactRecipe) error {
+func (s *Setup) installFromBuildResult(buildResult *model.BuildResult, artifacts map[artifact.ArtifactID]artifact.ArtifactRecipe) error {
 	var errors []error
 	wp := workerpool.New(MaxConcurrency)
 	downloads, err := artifact.NewDownloadsFromBuild(buildResult.BuildStatusResponse, buildResult.BuildEngine == model.Camel)
 	if err != nil {
 		return errs.Wrap(err, "Could not fetch artifacts to download.")
 	}
+	s.msgHandler.TotalArtifacts(len(downloads))
 	for _, a := range downloads {
 		func(a artifact.ArtifactDownload) {
 			wp.Submit(func() {
-				if err := s.setupArtifact(buildResult.BuildEngine, a.ArtifactID, a.UnsignedURI); err != nil {
+				// 'bundle' makes sense for the single camel download
+				name := "bundle"
```

Comment:
This is going to bite us. Please just feed in something that signals a painpoint, eg. "Unknown".

Reply:
Thinking about it again, I think I'll need to make this also build engine specific.  For camel it should be "bundle", for alternative builds "unknown".